SwiftCheck/Test.swift

```diff
@@ -6,6 +6,78 @@
 //  Copyright (c) 2015 TypeLift. All rights reserved.
 //
 
+// MARK: - Testing
+
+/// Property Testing is a more static and expressive form of Test-Driven Development that emphasizes
+/// the testability of program properties - A statement or invariant that can be proven to hold when
+/// fed any number of arguments of a particular kind.  It is akin to Fuzz Testing but is made
+/// significantly more power by the primitives in this framework.
+///
+/// A `Property` in SwiftCheck is more than just `true` and `false`, it is a value that is capable
+/// of producing a framework type called `Prop`, which models individual test cases that themselves
+/// are capable of passing or failing "in the small" with a `TestResult`.  This ability is encoded
+/// by a protocol called `Testable` that is adopted by Bool, Property, Prop, and several other
+/// internal framework types.  The thing to notice is the majority of user-facing functions and
+/// combinators in this library return types that conform to `Testable`.  This enables a high level
+/// of composition and enables the testing of incredibly complex properties that would normally be
+/// expressed by stateful, complected unit tests.
+///
+/// Below is the method all SwiftCheck properties are based on, `forAll`.  `forAll` acts as a
+/// "Quantifier", i.e. a contract that serves as a guarantee that a property holds when the given
+/// testing block returns `true` or truthy values, and fails when the testing block returns `false`
+/// or falsy values.  The testing block is usually used with Swift's abbreviated block syntax and
+/// requires type annotations for all value positions being requested.  For example,
+///
+///     forAll { (anInteger : Int, aBoolean : Bool, someStrings : ArrayOf<String>) in
+///         return true // This test case always passes.
+///     }
```

Comment:
Now the reader gets her very first taste of the system, and it's a bit confusing and not useful. Don't lead with a trivial `return true`. That only helps show syntax. It doesn't show any value. The very first example needs to be motivating. Why do I care? How does my not-too-big-not-too-small testing problem become really beautiful if I use SwiftCheck?

This also dumps in `ArrayOf<>` which feels like a typo rather than the awesome thing it really is. The next paragraph doesn't make it clear that `ArrayOf` is one of these magical Generators that are the whole point of this system.